firmware/ventilator-controller-stm32/Core/Test/Pufferfish/Protocols/crcelements.txt

```diff
@@ -0,0 +1,131 @@
+/* compute_body_crc
+Scenario: CRCElement: The compute_body_crc method correctly computes the CRC of a payload from a CRCElement body given as input
+  GIVEN("A CRC element initialized with an empty payload of capacity 250 bytes")
+    WHEN("the crc is computed on an input_buffer with empty payload")
+      THEN("the computed CRC is '0x00000000'")
+
+    WHEN("the crc is computed on an input_buffer with payload '123456789'")
+      THEN("the computed crc is '0xe3069283'")
+
+    WHEN("the CRC is computed on an input_buffer with payload '0x00' ")
+      THEN("the computed crc is '0x527d5351'")
+
+    WHEN("the crc is computed on an input_buffer with payload '0x01' ")
+      THEN("the computed crc is '0xa016d052'")
+
+/* write function
+Scenario: ConstructedCRCElement: The write method correctly generates a body from the payload given in the constructor
+  GIVEN("A crc element initalised with payload=[0x01, 0x02, 0x05]")
```

Comment:
I mentioned this in a previous comment but forgot to check whether it applied to other tests, but it does: let's use "constructed" instead of "initialized" when we refer to inputs given into the constructor. There might rarely be some tests of stateful classes where we use both, e.g. "GIVEN a StateSynchronizer constructed with ____ schedules/etc. and initialized to be on the 1st schedule entry"

Reply:
Fixed.